tools/src/main/scala/scala/scalanative/build/Config.scala

```diff
@@ -150,7 +193,18 @@ object Config {
     override lazy val workdir: Path =
       basedir.resolve(s"native$nameSuffix")
 
-    override lazy val artifactPath: Path = {
+    override lazy val basename: String =
+      if (compilerConfig.multipleMains &&
+          compilerConfig.buildTarget == BuildTarget.application) {
+        mainClass.get
```

Comment:
It might work but would lead to very long fully qualified names, because of that user might want to override the artifact name, but that's impossible since we calculate it. 

I wonder, what if we would have internal and external artifacts:
* internal artifact is a file stored in our `workdir` ("native" directory in the target). We could potentially store there raw artifacts, e.g. named as mainClass or `moduleName` + hash of mainClass and possibly other settings. After successful linking we would copy it to the external artifact path (copy instead of symlink because it's broken on Windows and makes it easier to share the output) 
* the external ones are based on the module name, its names does not change between subsequent links.
What do you think? 

Also because we have our workdir we can store store other significant information required for detecting if we can potentially skip the build (storage is cheap, read from local IO is relatively cheap)

Reply:
I like that idea to use `moduleName` + hash of main and then the copy. We can certainly go with storing info about the build settings and caching and stuff but that can be done later as it shouldn't affect the API. Denys suggested something like this as well.